lavuelib/imageSource.py

```diff
@@ -361,43 +359,31 @@ def getData(self):
         try:
             image = None
             metadata = ""
+            frame = 0
             try:
                 if self.__handler is None:
                     self.__handler = imageFileHandler.NexusFieldHandler(
                         str(self.__nxsfile))
                 if self.__node is None:
                     self.__node = self.__handler.getNode(self.__nxsfield)
-                    metadata = self.__handler.getMetaData(self.__node)
+                    try:
+                        metadata = self.__handler.getMetaData(self.__node)
+                    except Exception:
+                        metadata = ""
                     # if metadata:
                     #     print("IMAGE Metadata = %s" % str(metadata))
                 fid = self.__handler.getLastFrame(self.__node, self.__gdim)
-                if self.__nxslast:
-                    if fid - 1 > self.__frame or fid - 1 < self.__frame:
-                        self.__frame = fid - 1
+                if not self.__frame == -1:
+                    if fid < self.__frame:
+                        self.__frame = fid
+                    frame = self.__frame
                 else:
-                    if fid - 1 < self.__frame:
-                        self.__frame = fid - 1
+                    frame = fid
 
                 image = self.__handler.getImage(
                     self.__node, self.__frame, self.__gdim)
-            # except Exception:
             except Exception:
-                try:
```

Comment:
This part of the code is largely duplicated with L364 to L383. I found the only difference between these two parts is about the metadata, so I refactored it.

Reply:
Yes, you are right. Thanks.